samples/add-features-with-contingent-values/src/main/java/com/esri/arcgismaps/sample/addfeatureswithcontingentvalues/components/MapViewModel.kt

```diff
@@ -0,0 +1,383 @@
+            featureTable?.getContingentValuesOrNull(it, "BufferSize")
+        }
+
+        val bufferSizeRangeValue = contingentValueResult?.byFieldGroup?.get("BufferSizeFieldGroup")
+            ?.get(0) as ContingentRangeValue
+        val minValue = bufferSizeRangeValue.minValue as Int
+        val maxValue = bufferSizeRangeValue.maxValue as Int
+
+        val sliderControlParameters = if (maxValue > 0) {
+            SliderControlParameters(true, minValue, maxValue)
+        } else {
+            SliderControlParameters()
+        }
+
+        return sliderControlParameters
+    }
+
+    fun onBufferSizeSelect(bufferSize: Int) {
+        feature?.attributes?.set("BufferSize", bufferSize)
+        _featureEditState.value = _featureEditState.value.copy(buffer = bufferSize)
+    }
+
+    /**
+     * Ensure that the selected values are a valid combination.
+     * If contingencies are valid, then display [feature] on the [mapPoint]
+     */
+    fun validateContingency(mapPoint: Point) {
+        // check if all the features have been set
+        if (featureTable == null) {
+            messageDialogVM.showMessageDialog("Input all values to add a feature to the map")
+            return
+        }
+
+        // validate the feature's contingencies
+        val contingencyViolations = feature?.let {
+            featureTable?.validateContingencyConstraints(it)
+        } ?: return messageDialogVM.showMessageDialog("No feature attribute was selected")
+
+        // if there are no contingency violations
+        if (contingencyViolations.isEmpty()) {
+            // the feature is valid and ready to add to the feature table
+            // create a symbol to represent a bird's nest
+            val symbol = SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Circle, Color.black, 11F)
+            // add the graphic to the graphics overlay
+            graphicsOverlay.graphics.add(Graphic(mapPoint, symbol))
+
+            // set the geometry of the feature to the map point
+            feature?.geometry = mapPoint
+
+            // create the graphic of the feature
+            val graphic = feature?.let { createGraphic(it) }
+            // add the graphic to the graphics overlay
+            graphic?.let { graphicsOverlay.graphics.add(it) }
+
+            // add the feature to the feature table
+            viewModelScope.launch {
+                feature?.let { featureTable?.addFeature(it) }
+                feature?.load()?.getOrElse {
+                    return@launch messageDialogVM.showMessageDialog("Error", it.message.toString())
+                }
+            }
+        } else {
+            val violations = contingencyViolations.joinToString(separator = "\n") {
+                violation -> violation.fieldGroup.name
+            }
+            messageDialogVM.showMessageDialog(
+                "Invalid contingent values",
+                "${contingencyViolations.size} violations found:\n" + violations
+            )
```

Comment:
We could associate the `ContingencyConstraintViolation` field groups with UI elements & restyle them accordingly. It could be fairly time consuming, and is more of a showcase of Compose than of our SDK.